### PR TITLE
ACP 77 new commands

### PR DESCRIPTION
**Pull request by sukantoraymond**
## Why this should be merged
Convert Subnet Tx
Register Validator Tx
Change Weight of Validator
Remove Validator

Commands modified:
avalanche deploy (adds ConvertSubnetTx to the command)
avalanche addValidator (use flag --not-Sov for legacy add Validator, otherwise uses registerSubnetTx)
avalanche convert (converts legacy subnet into sov)
avalanche removeValidator  (use flag --not-Sov for legacy add Validator, otherwise uses setSubnetWeight)
avalanche changeWeight 

## How this works
--pending avalanchego impelmentation--
./bin/avalanche subnet deploy <subnetName>

### CODE REVIEW

File: cmd/blockchaincmd/add_validator.go [hunk 1]
```diff
@@ -64,33 +72,31 @@ these prompts by providing the values with flags.
...
-
-	cmd.Flags().StringSliceVar(&subnetAuthKeys, "subnet-auth-keys", nil, "control keys that will be used to authenticate add validator tx")
-	cmd.Flags().StringVar(&outputTxPath, "output-tx-path", "", "file path of the add validator tx")
+	cmd.Flags().Uint64Var(&weight, "weight", constants.BootstrapValidatorWeight, "set the staking weight of the validator to add")
+	cmd.Flags().Uint64Var(&balance, "balance", 0, "set the AVAX balance of the validator that will be used for continuous fee to P-Chain")
 	cmd.Flags().BoolVarP(&useEwoq, "ewoq", "e", false, "use ewoq key [fuji/devnet only]")
 	cmd.Flags().BoolVarP(&useLedger, "ledger", "g", false, "use ledger instead of key (always true on mainnet, defaults to false on fuji/devnet)")
 	cmd.Flags().StringSliceVar(&ledgerAddresses, "ledger-addrs", []string{}, "use the given ledger addresses")
-	cmd.Flags().BoolVar(&waitForTxAcceptance, "wait-for-tx-acceptance", true, "just issue the add validator tx, without waiting for its acceptance")
+	cmd.Flags().BoolVar(&nonSOV, "not-sov", false, "set to true if adding validator to a non-SOV blockchain")
```
Comment (arturrez) [inline cmd/blockchaincmd/add_validator.go:85]: why do we use inverse logic here?

File: cmd/blockchaincmd/add_validator.go [hunk 2]
```diff
@@ -64,33 +72,31 @@ these prompts by providing the values with flags.
...
+	cmd.Flags().StringVar(&publicKey, "public-key", "", "set the BLS public key of the validator to add")
```
Comment (arturrez) [inline cmd/blockchaincmd/add_validator.go:86]: nit: i think these params should start with bls, like `--bls-public-key`  and `--bls-proof-of-possetion`. please feel free to ignore this comment

File: pkg/prompts/validations.go [hunk 3]
```diff
@@ -129,13 +129,13 @@ func validateWeight(input string) error {
...
 
-func validateBootstrapValidatorBalance(input string) error {
+func validateValidatorBalance(input string) error {
 	val, err := strconv.ParseUint(input, 10, 64)
 	if err != nil {
 		return err
 	}
 	if val == 0 {
-		return fmt.Errorf("initial bootstrap validator balance must be greater than 0 AVAX")
+		return fmt.Errorf("subnet validator balance must be greater than 0 AVAX")
```
Comment (felipemadero) [inline pkg/prompts/validations.go:138]: I vote to ask for > 1 AVAX

**Review by arturrez — APPROVED** [review]

File: cmd/blockchaincmd/change_weight.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+		ux.Logger.PrintToUser("failed to check if node is a validator on the subnet: %s", err)
+	} else if !isValidator {
+		// this is actually an error
+		return fmt.Errorf("node %s is not a validator on subnet %s", nodeID, subnetID)
+	}
+
+	deployer := subnet.NewPublicDeployer(app, kc, network)
+
+	// first remove the validator from subnet
+	err = removeValidatorSOV(deployer, network, subnetID, nodeID)
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/blockchaincmd/change_weight.go:146]: yes, previous to adding, but after asking all questions

File: cmd/blockchaincmd/remove_validator.go [hunk 5]
```diff
@@ -85,15 +99,23 @@ func removeValidator(_ *cobra.Command, args []string) error {
...
+	_, err = ValidateSubnetNameAndGetChains([]string{blockchainName})
 	if err != nil {
 		return err
 	}
-	blockchainName := chains[0]
 
 	switch network.Kind {
 	case models.Local:
-		return removeFromLocal(blockchainName)
+		if nonSOV {
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/blockchaincmd/remove_validator.go]: this should be more general, but that is for another PR

**Pull request merged**